src/Security/Authentication/Twitter/src/TwitterHandler.cs

```diff
@@ -318,5 +326,79 @@ private string ComputeSignature(string consumerSecret, string tokenSecret, strin
                 return Convert.ToBase64String(hash);
             }
         }
+
+        private async Task EnsureTwitterRequestSuccess(HttpResponseMessage response)
+        {
+            var contentTypeIsJson = Equals(response.Content.Headers.ContentType, new MediaTypeHeaderValue("application/json"));
+            var contentTypeIsXml = Equals(response.Content.Headers.ContentType, new MediaTypeHeaderValue("application/xml"));
+
+            if (!response.IsSuccessStatusCode &&
```

Comment:
Style: you can reduce the nesting by inverting this condition and returning early for the negative case.